src/test/java/wolox/training/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,168 @@
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(controllers = UsersController.class)
+class UserControllerTest {
+
+    @Autowired
+    private MockMvc mvc;
+
+    @MockBean
+    private UsersRepository mockUsersRepository;
+    @MockBean
+    private BookRepository mockBookRepository;
+
+    private static User oneTestUser;
+    private static User twoTestUser;
+    private static Book oneTestBook;
+    private static List<User> manyTestUsers;
+    private static List<Book> manyTestBooks;
+    private static final String USER_PATH = "/api/users";
+
+    @BeforeAll
+    static void setUp() {
+        oneTestUser = TestEntities.mockOneUser();
+        manyTestUsers = TestEntities.mockManyUsers();
+        oneTestBook = TestEntities.mockBook();
+        twoTestUser = TestEntities.mockTwoUser();
+        manyTestBooks = new ArrayList<>();
+        manyTestBooks.add(oneTestBook);
+        twoTestUser.setBooks(manyTestBooks);
+    }
+
+    @Test
+    @DisplayName("Test find all user ,return status OK")
+    void whenFindUserByIdThenReturnStatusOK() throws Exception {
+        given(mockUsersRepository.findById(1L)).willReturn(Optional.of(oneTestUser));
+        String url = (USER_PATH + "/1");
+        mvc.perform(get(url)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.name").value(oneTestUser.getName()));
+
+    }
+
+    @Test
+    @DisplayName("Test, When a user is searched for its id,it return status not found")
+    void whenUserThatNotExistsThenReturnNotFound() throws Exception {
+        given(mockUsersRepository.findById(1L)).willReturn(Optional.empty());
+        String url = (USER_PATH + "/1");
+        mvc.perform(get(url)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNotFound());
+    }
+
+    @Test
+    @DisplayName("Test,When a users is searched ,it return status OK")
+    void whenFindAllUserTHenReturnStatusOK() throws Exception {
+        given(mockUsersRepository.findAll()).willReturn(manyTestUsers);
+        String url = USER_PATH;
+        mvc.perform(get(url)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    @DisplayName("Test , When a user is created , it return status Created")
+    void whenCreateUserThenReturnStatusCreated() throws Exception {
+        String json = new ObjectMapper().writeValueAsString(oneTestUser);
+        String url = USER_PATH;
+        mvc.perform(post(url)
+                .contentType(MediaType.APPLICATION_JSON)
+                .characterEncoding("utf-8")
+                .content(json))
+                .andDo(print())
+                .andExpect(status().isCreated());
+    }
+
+    @Test
+    @DisplayName("Test, When a user is updated , it return status OK")
+    void whenUpdateUserThenReturnStatusCreated() throws Exception {
```

Comment:
Can you test please when update an user that does not exists and returns not found?